lib/new_relic/agent/log_event_aggregator.rb

```diff
@@ -58,38 +59,68 @@ def capacity
       end
 
       def record(formatted_message, severity)
-        return unless enabled?
+        return unless logger_enabled?
 
         severity = 'UNKNOWN' if severity.nil? || severity.empty?
+        increment_event_counters(severity)
 
-        if NewRelic::Agent.config[METRICS_ENABLED_KEY]
```

Comment:
A few blocks of code were shared by `create_event` and `create_logstasher_event`. This is an example of a block of code moved into a new method (`increment_event_counters` is the name for this)